lib/ansible/modules/cloud/google/gce_labels.py

```diff
@@ -0,0 +1,315 @@
+
+    return {
+        'resource_name': resource.get('name'),
+        'resource_url': resource.get('selfLink'),
+        'resource_type': resource_type,
+        'resource_location': resource_location,
+        'label_fingerprint': resource.get('labelFingerprint'),
+        'labels': resource.get('labels')
+    }
+
+
+def _set_labels(client, ri, params):
+    result = err = None
+    labels = {
+        'labels': params['labels'],
+        'labelFingerprint': ri['label_fingerprint']
+    }
+
+    # TODO(all): See the comment above for KNOWN_RESOURCES. As labels are
+    # added to the v1 GCE API for more resources, some minor code work will
+    # need to be added here.
+    if ri['resource_type'] == 'instances':
+        req = client.instances().setLabels(project=params['project_id'],
+                                           instance=ri['resource_name'],
+                                           zone=ri['resource_location'],
+                                           body=labels)
+    elif ri['resource_type'] == 'disks':
+        req = client.disks().setLabels(project=params['project_id'],
+                                       zone=ri['resource_location'],
+                                       resource=ri['resource_name'],
+                                       body=labels)
+    elif ri['resource_type'] == 'snapshots':
+        req = client.snapshots().setLabels(project=params['project_id'],
+                                           resource=ri['resource_name'],
+                                           body=labels)
+    elif ri['resource_type'] == 'images':
+        req = client.images().setLabels(project=params['project_id'],
+                                        resource=ri['resource_name'],
+                                        body=labels)
+    else:
+        raise ValueError('Unsupported resource type: %s' % ri['resource_type'])
+
+    # TODO(erjohnso): Once Labels goes GA, we'll be able to use the GCPUtils
+    # method to poll for the async request/operation to complete before
+    # returning. However, during 'beta', we are in an odd state where
+    # API requests must be sent to the 'compute/beta' API, but the python
+    # client library only allows for *Operations.get() requests to be
+    # sent to 'compute/v1' API. The response operation is in the 'beta'
+    # API-scope, but the client library cannot find the operation (404).
+    # result = GCPUtils.execute_api_client_req(req, client=client, raw=False)
+    # return result, err
+    result = req.execute()
+    return True, err
+
+
+def main():
+    module = AnsibleModule(argument_spec=dict(
+        state=dict(choices=['absent', 'present'], default='present'),
+        service_account_email=dict(),
+        service_account_permissions=dict(type='list'),
+        pem_file=dict(),
+        credentials_file=dict(),
+        labels=dict(required=False, type='dict'),
+        resource_url=dict(required=False, type='str'),
```

Comment:
In the module args, you can set certain arguments as mutually exclusive so users don't specify them together. For this module I think resource_url should be exclusive with the _name, _location, and _type options.